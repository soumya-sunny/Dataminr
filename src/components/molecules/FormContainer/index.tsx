import React from 'react'
import Paper from '@mui/material/Paper';
import { IForm,ToggleState } from '../../../types/types';
import styled from 'styled-components';
import {  Grid, Switch } from '@mui/material';


interface IFormContainerProps {
  data: IForm;
}

const StyledPaper = styled(Paper)`
padding:10px;
margin:10px;
`
// TODO make proper controlled component for number field
const FormContainer: React.FC<IFormContainerProps> = ({ data }) => {
  const { label, inputs } = data;
  return (
    <StyledPaper>
      <Grid container justifyContent="space-between">
        <Grid item sx={{ alignSelf: 'flex-start' }}>{label.toUpperCase()}</Grid>
        <Grid item sx={{ alignSelf: 'flex-end' }}>
          {inputs.map((input,i) =>(
            input.type === "number" ? <input type="number" value={input.value}  key={i} /> :
             <Switch defaultChecked={input.value===ToggleState.ON} key={i} />
            ))}
        </Grid>
      </Grid>
    </StyledPaper>
  )
}

export default FormContainer