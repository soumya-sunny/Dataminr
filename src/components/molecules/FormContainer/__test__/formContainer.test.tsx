import React from 'react';
import { render, screen } from '@testing-library/react';
import FormContainer from '../index';

const data = {
    "label": "Audit Log",
    "inputs": [
        {
            "type": "number",
            "value": "10"
        },
        {
            "type": "toggle",
            "value": "on"
        }
    ]
};
describe(' FormContainer Test Suite', () => {
    it('should match with snapshot', () => {
        const { asFragment } = render(<FormContainer data={data} />);
        expect(asFragment()).toMatchSnapshot();
    })
    it('should have number field when type is number', () => {
         render(<FormContainer data={data} />);
         const input = screen.getByRole('spinbutton');
        expect(input).toBeInTheDocument();
    })
    
// TODO test for label 
});
