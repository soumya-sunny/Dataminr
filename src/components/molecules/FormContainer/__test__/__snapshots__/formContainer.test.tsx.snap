// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` FormContainer Test Suite should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-bdvvtL isMKnI css-1ps6pg7-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-1abg2ox-MuiGrid-root"
      >
        AUDIT LOG
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-1d2j7l7-MuiGrid-root"
      >
        <input
          type="number"
          value="10"
        />
        <span
          class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
        >
          <span
            class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
